docs/spec/sources.md

```diff
@@ -0,0 +1,338 @@
+# Sources
+
+A **source** is any resource that generates or imports an event and relays that
+event to another resource on the cluster via CloudEvents. Sourcing events is
+critical to developing a distributed system that reacts to events.
+
+A Source:
+
+- Represents an off or on-cluster system, service or application that produces
+  events to be consumed by a sink.
+- Produces or imports CloudEvents.
+- Sends CloudEvents to the configured **sink**.
```

Comment:
At some point we will have to say something about error handling: what happens when sink failed to receive events.

Reply:
I am not sure the source spec can say. It depends on how durable you are and need to be.